tests/foreman/longrun/test_puppet.py

```diff
@@ -0,0 +1,89 @@
+"""End to end test for Puppet funcionality
+
+@Requirement: Puppet
+
+@CaseAutomation: Automated
+
+@CaseLevel: Acceptance
+
+@CaseComponent: CLI
+
+@TestType: Functional
+
+@CaseImportance: High
+
+@Upstream: No
+"""
+
+from robottelo.config import settings
+from robottelo.decorators import (
+    run_in_one_thread,
+    run_only_on,
+    skip_if_not_set,
+    stubbed,
+    tier4,
+)
+from robottelo.test import CLITestCase
+
+
+@run_in_one_thread
+@skip_if_not_set('clients')
+class PuppetScenarioTestCase(CLITestCase):
+    """Implements Puppet test scenario"""
+
+    @classmethod
+    def setUpClass(cls):
+        """ Spin up Capsules and Clients.
+        """
+        super(PuppetScenarioTestCase, cls).setUpClass()
+        cls.sat6_hostname = settings.server.hostname
+        """
+        1. prepare for spinning up VMs (spinned up along testing)
+        """
```

Comment:
I think you added it as ToDo. So can be added like:`#  Todo:  prepare for spinning up VMs (spinned up along testing)`